packages/vite/src/node/optimizer/scan.ts

```diff
@@ -219,6 +222,16 @@ async function prepareEsbuildScanner(
     format: 'esm',
     logLevel: 'silent',
     plugins: [...plugins, plugin],
+    tsconfigRaw:
+      typeof tsconfigRaw === 'string'
+        ? tsconfigRaw
+        : {
+            ...tsconfigRaw,
+            compilerOptions: {
+              experimentalDecorators: true,
+              ...tsconfigRaw?.compilerOptions,
+            },
+          },
```

Comment:
I wonder if we should use:
```suggestion
    tsconfigRaw:
      tsconfigRaw ?? {
        compilerOptions: {
          experimentalDecorators: true,
        },
      },
    },
```
so it works consistently for the three cases, `experimentalDecorators` is only set if the user doesn't manually modify the config.
- `tsconfigRaw` is an object
- `tsconfigRaw` is a string
- `tsconfig` passed as a path

Right now, only the object form is respected. Or the other way around, the file or string could be first converted to an object.

I'm fine moving forward with the current patch though if we consider it as a temporal solution to help the majority of users. We could add a comment here in that case

Reply:
Wouldn't that not apply the `experimentalDecorators: true` default if the user pass something like `{ compilerOptions: { useDefineForClassFields: true } }`. Currently in `transformWithEsbuild` having that config would fallback with `experimentalDecorators: true` too. I think for string form it's fine to respect the user input completely like `transformWithEsbuild` does for now.

Reply:
Oh you're right that `tsconfigRaw` would've overwrite `tsconfig` if only `tsconfig` is set (iirc) 🤔 I think we can not set `tsconfigRaw` if `tsconfig` is specified?

Reply:
Yes, it won't apply it in that case. I find it inconsistent to only do it for the object form, so I was trying to see what would be correct here. But I was thinking along the lines of what @sapphi-red said. Let's merge this one as is for now then.

Reply:
Oh, it seems an error happens. [esbuild repl](https://esbuild.github.io/try/#YgAwLjE4LjEyAC0tdHNjb25maWctcmF3PSJ7fSIgLS10c2NvbmZpZz0idHNjb25maWcuanNvbiIAZQBlbnRyeS50cwBAZm9vCmNsYXNzIEZvbyB7Cn0AAHRzY29uZmlnLmpzb24Ae30)
So we need to skip adding `tsconfigRaw` if `tsconfig` exists.

Reply:
Ah ok I can fix that in a followup PR now.